linera-views/src/journaling.rs

```diff
@@ -230,72 +234,73 @@ where
     /// Writes blocks to the database and resolves them later.
     async fn write_journal(
         &self,
-        simplified_batch: K::SimpBatch,
+        simplified_batch: K::Batch,
         base_key: &[u8],
     ) -> Result<JournalHeader, K::Error> {
         let mut iter = simplified_batch.into_iter();
-        let mut value_size = 0;
-        let mut transact_size = 0;
-        let mut simp_batch = K::SimpBatch::default();
+        let mut block = K::Batch::default();
+        let mut block_size = 0;
         let mut block_count = 0;
-        let mut transacts = K::SimpBatch::default();
+        let mut transaction = K::Batch::default();
+        let mut transaction_size = 0;
         loop {
-            let result = simp_batch.try_append(&mut iter, &mut value_size)?;
-            if !result {
+            if !iter.write_next_value(&mut block, &mut block_size)? {
                 break;
             }
-            let (value_flush, transact_flush) = if (iter.remaining_len() == 0)
-                || simp_batch.len() == K::MAX_TRANSACT_WRITE_ITEM_SIZE - 2
-            {
-                (true, true)
-            } else {
-                let next_value_size = simp_batch.next_value_size(value_size, &mut iter)?;
-                let next_transact_size = transact_size + next_value_size;
-                let value_flush = if next_transact_size > K::MAX_TRANSACT_WRITE_ITEM_TOTAL_SIZE {
-                    true
+            let (block_flush, transaction_flush) = {
+                if iter.is_empty() || block.len() == K::MAX_BATCH_SIZE - 2 {
+                    (true, true)
                 } else {
-                    next_value_size > K::MAX_VALUE_SIZE
-                };
-                let transact_flush = next_transact_size > K::MAX_TRANSACT_WRITE_ITEM_TOTAL_SIZE;
-                (value_flush, transact_flush)
+                    let next_block_size = iter.next_batch_size(&block, block_size)?;
+                    let next_transaction_size = transaction_size + next_block_size;
+                    let block_flush = if next_transaction_size > K::MAX_BATCH_TOTAL_SIZE {
+                        true
+                    } else {
+                        next_block_size > K::MAX_VALUE_SIZE
+                    };
+                    let transaction_flush = next_transaction_size > K::MAX_BATCH_TOTAL_SIZE;
+                    (block_flush, transaction_flush)
+                }
             };
-            if value_flush {
-                value_size += simp_batch.overhead_size();
-                let value = simp_batch.to_bytes()?;
-                assert_eq!(value.len(), value_size);
+            if block_flush {
+                block_size += block.overhead_size();
+                let value = bcs::to_bytes(&block)?;
+                block = K::Batch::default();
```

Comment:
It is not great to use the word `block` for something that is a `batch`.

Reply:
Yes, it could be `block_batch` and `transaction_batch`.